Sync

yeqbfgxjiq opened the pull request:
It works. Currently it's only syncing to the Redemptions `overview.md` doc, but we can add more as we get the other apps ready. Note: every time we want the docs to be updated on the website we just have to run the publish script again, and now built into publishing it'll fetch the latest docs from wherever the sync script is pointed to.

yeqbfgxjiq committed:
created bug-bounty.md

yeqbfgxjiq committed:
setup basic sync scripts

yeqbfgxjiq committed:
added dependencies to package.json

yeqbfgxjiq committed:
removed extra package.json in root dir

lkngtn commented:
I tried to preview this locally, but had trouble getting it to run. `yarn start` fails with the following:

```
$ npm run clean:docs && npm run sync:dandelion-apps

> @ clean:docs /Users/luke/repos/1hive/website/website
> rm -r ../docs/sync/*

rm: ../docs/sync/*: No such file or directory
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ clean:docs: `rm -r ../docs/sync/*`
npm ERR! Exit status 1
```

lkngtn commented:
using the `-f` flag will resolve the error, but the redemptions file also doesn't seem to be syncing.

lkngtn commented on website/package.json:
```diff
@@ -1,14 +1,23 @@
     "examples": "docusaurus-examples",
-    "start": "docusaurus-start",
-    "build": "docusaurus-build",
+		"clean:docs": "rm -r ../docs/sync/*",
```
Needs to be `rm -rf ../docs/sync/*`

yeqbfgxjiq replied:
Oh great! Yeah I would just add a placeholder file in there for `rm -r` to remove, but `-rf` seems to work too

yeqbfgxjiq replied:
Just pushed that to the PR

lkngtn commented on website/yarn-error.log:
```diff
@@ -0,0 +1,6720 @@
+Arguments: 
```
maybe this should be added to .gitignore?

yeqbfgxjiq replied:
Good call

lkngtn commented on website/scripts/sync-dandelion-apps.js:
```diff
@@ -0,0 +1,28 @@
+const { syncPages } = require('./sync-util')
+
+const GIT_REF = 'master'
```
This seems like it will only work for one repo?

yeqbfgxjiq replied:
The sync utility works for an arbitrary amount of repos. To sync multiple sources to the webiste, just create a new sync file `sync-newstuff.js` for every repo and point it to whatever you want to sync.

lkngtn reviewed (changes requested):
I wasn't able to get this to work, it doesn't seem the sync function is syncing files correctly, so the redemptions page is missing/not-updated to the file in the redemptions repo.

lkngtn committed:
Merge pull request #45 from 1Hive/removing-extra-package.json

removed extra package.json in root dir

lkngtn committed:
fixed broken links in dandelion org overview

onbjerg committed:
Revive dead links in "Roles and Responsibilities"

lkngtn committed:
Merge pull request #49 from 1Hive/fix-dead-links

Revive dead links in "Roles and Responsibilities"

lkngtn committed:
Merge pull request #47 from 1Hive/lkngtn

fixed broken links in dandelion org overview page

lkngtn committed:
Changing the url for publishing is not necessary

You don't need to make any changes to view locally anymore `yarn start` should just work, and not changing it each time for deployments seems simpler/safer.

lkngtn committed:
Merge pull request #43 from 1Hive/bug-bounty

created bug-bounty.md

onbjerg committed:
Fix broken link to "Projects and Tasks"

onbjerg committed:
Fix broken "Learn more" link

yeqbfgxjiq committed:
Merge pull request #51 from 1Hive/broken-learn-more

Fix broken "Learn more" link

yeqbfgxjiq committed:
Merge pull request #50 from 1Hive/project-tasks-broken-link

Fix broken link to "Projects and Tasks"

burrrata commented:
Hmmm interesting. It totally works for multiple repos/sync files and I've deployed it on multiple Docusauruses for multiple projects. It might be an error where if the docs /sync folder is empty, it technically can't be cleaned so you need to have some placeholder files in there for the thing to work. I'll check it out now to get it working

yeqbfgxjiq committed:
Merge branch 'sync' of https://github.com/1Hive/website into sync

yeqbfgxjiq committed:
works on my machine! lol

yeqbfgxjiq committed:
updated clean script to force remove the sync directory so that we dont need placeholder files anymore

yeqbfgxjiq committed:
added yarn-error.log to .gitignore

burrrata commented:
@lkngtn this PR is kind of getting nebulous. Lots of things in here beyond just the sync scripts. Let's review and merge this asap before more stuff gets piled on lol

burrrata commented:
@lkngtn marked the change requests as resolved because I added the fixes to the PRs, but now that I think about it are you supposed to close those or am I? lol

burrrata commented:
@lkngtn last time I checked this was all working on my machine. Was there anything more you wanted to review or should I go ahead and merge it all in?
